### PR TITLE
Cross-compile traefik for Windows

### PR DESCRIPTION
**Pull request by sodre**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by bollwyvl** [comment]
${DIETY}speed to you! If you are interested in this as a long-term maintenance goal, please consider adding yourself as a maintainer.

You may well find a lot of... surprises in getting the machinery to work correctly: the `go` stdlib libraries usually show up incorrectly in the license checker, and may need manual `--ignore` flags (see [prior art](https://github.com/search?q=org%3Aconda-forge+go+licenses+%2F--ignore%2F&type=code)).

File: recipe/meta.yaml [hunk 1]
```diff
@@ -22,9 +21,9 @@ requirements:
 
 test:
   commands:
-    - traefik version
-    - traefik version | grep -iE  "Version:\s+{{ version.replace(".", "\.") }}$"
-    - traefik --help
+    - traefik{{ target_goexe }} version
+    - traefik{{ target_goexe }} version | grep -iE  "Version:\s+{{ version.replace(".", "\.") }}$"
+    - traefik{{ target_goexe }} --help
```
Comment (sodre) [inline recipe/meta.yaml:26]: These just expand to an empty string in Unix builds, and ".exe" on windows.

File: conda-forge.yml [hunk 2]
```diff
@@ -2,6 +2,7 @@ build_platform:
   osx_arm64: osx_64
   linux_aarch64: linux_64
   linux_ppc64le: linux_64
+  win_64: linux_64
```
Comment (sodre) [inline conda-forge.yml:5]: @bollwyvl , this is the main change that forces the cross-compile for windows.

**Comment by bollwyvl** [comment]
Again, please consider adding yourself as a maintainer. I have very little experience with `go` cross-compiling, and don't use windows (much less traefik on windows), and am concerned what I would do when it breaks.

**Comment by sodre** [comment]
> tle experience with `go` cross-compiling, and don't use windows (much less traefik on windows), and am concerned what I would

Done :)

**Pull request merged**